src/cogent3/core/new_alignment.py

```diff
@@ -356,35 +395,72 @@ def take_seqs(
         negate
             select all sequences EXCEPT names
         kwargs
-            keyword arguments to be passed to the constructor of the new seqcollection
+            keyword arguments to be passed to the constructor of the new collection
+
+        Notes
+        -----
+        The seqs in the new collection will be references to the same objects as
+        the seqs in the old collection.
         """
+        # todo: kath, note gavins comment that this method could operate on only the names
+        # list and not the seqs_data object
 
-        # todo: kath, add subset method onto SeqsData
         if negate:
-            selected_seqs = {
-                name: self.seqs.get_seq_str(seqid=name)
-                for name in self.names
-                if name not in names
-            }
+            names = [name for name in self.names if name not in names]
         else:
-            selected_seqs = {
-                name: self.seqs.get_seq_str(seqid=name)
-                for name in names
-                if name in self.names
-            }
+            names = [name for name in names if name in self.names]
 
-        if not selected_seqs:
+        if not names:
             return {}
 
-        moltype = kwargs.pop("moltype", self.moltype)
+        seqs_data = self.seqs.subset(names)
 
-        seqs_data = SeqsData(data=selected_seqs, alphabet=moltype.alphabet)
+        moltype = kwargs.pop("moltype", self.moltype)
         result = self.__class__(
             seqs_data=seqs_data, moltype=moltype, info=self.info, **kwargs
         )
-        # todo: kath, update annotation_db and assign to result when we have annotations working
+        if self.annotation_db:
```

Comment:
copying an annotation_db could be quite an expensive operation. Let's add an optional argument `copy_annotations=False`. In that case we just assign the same annotation db.